hicommonwealth/commonwealth
# Deposit natural denom

## Link to Issue
Closes: #4922

## Description of Changes
- Adds util functions to convert minimalDenom amount to/from naturalDenom amount
  - Ex: 8000000uosmo -> 8 OSMO
  - 7 EVMOS -> 7000000000000000000aevmos
- Adds a ETHERMINT_CHAINS constant to set those to 18 decimals (was at 6 before)
- Displays deposit input in _DENOM_
- converts user input to _udenom_ for transaction
- Updates ethermint devnet bootstrap.sh to use a more sane minDeposit of 20 EVMOS

## "How We Fixed It"


## Test Plan
- Added unit tests for conversion functions
- CA (click around) tested on local and frack:
  - Using shared account, join /csdk-beta community
  - Click "+" to make new onchain proposal, Choose "Community Spend"
    - Confirm Deposit field defaults to 2 STAKE
    - Confirm Amount field is denominated in STAKE
  - Create a proposal with `cosmos18q3tlnx8vguv2fadqslm7x59ejauvsmnlycckg` as recipient, and **0.000005** STAKE as the "Amount." Submit with keplr.
    - When the TX goes through, confirm that the processed spend amount is correct by expanding the data object in the "Community Spend Proposal" section of the proposal page
      - Confirm the amount is 5 ustake:
      - <img width="732" alt="Screenshot 2023-09-07 at 4 27 28 PM" src="https://github.com/hicommonwealth/commonwealth/assets/9438198/dddfd4c0-5d83-483e-b930-97aab1e769ad">

  - check `/evmos/new/proposal` (3500 EVMOS) or `/evmos-dev/new/proposal` (20 EVMOS) to confirm correct denoms

## Deployment Plan

1. 

## Other Considerations

- Most Cosmos chains use 6 decimal places for their minimal denom. Ethermint chains use 18.
